pandas/core/construction.py

```diff
@@ -358,15 +358,17 @@ def array(
             return cls._from_sequence(data, dtype=dtype, copy=copy)
 
         elif data.dtype.kind in "iu":
-            return IntegerArray._from_sequence(data, copy=copy)
+            dtype = IntegerArray._dtype_cls._get_dtype_mapping()[data.dtype]
```

Comment:
```suggestion
            dtype = NUMPY_INT_TO_DTYPE[data.dtype]
```

?

Reply:
thatd work too.  i chose to do it this way to avoid the extra import